save-demo/src/main/kotlin/com/saveourtool/save/demo/storage/ToolStorage.kt

```diff
@@ -20,34 +16,13 @@ private const val TOOLS_PATH = "tools"
 class ToolStorage(
     configProperties: ConfigProperties,
 ) : AbstractFileBasedStorage<ToolKey>(Path.of(configProperties.fileStorage.location) / TOOLS_PATH) {
-    /**
-     * Todo: Implement DownloadToolService
-     */
-    @PostConstruct
-    fun downloadTools() {
-        val supportedTools = listOf(
-            ToolKey("diktat", "1.2.3", "diktat"),
-            ToolKey("diktat", "1.2.3", "diktat.cmd"),
-            ToolKey("ktlint", "0.47.1", "ktlint-cli"),
-            ToolKey("ktlint", "0.47.1", "ktlint-cli.cmd"),
-        )
-        list().collectList()
-            .flatMapIterable { availableFiles ->
-                supportedTools.filter {
-                    it !in availableFiles
-                }
-            }
-            .flatMap { key ->
-                upload(key, ClassPathResource(key.executableName).toByteBufferFlux()).thenReturn(key)
-            }
-            .subscribe()
-    }
-
+    @Suppress("COMPLEX_EXPRESSION")
     override fun buildKey(rootDir: Path, pathToContent: Path): ToolKey = ToolKey(
+        pathToContent.parent.parent.parent.name,
```

Comment:
Please extract `pathToContent.parent` and `pathToContent.parent.parent` into local variables.

Reply:
To my mind there are two possible solutions:
1. Leave as is as far as it is a kind of logic that is very internal (probably add tests for it)
2. Move ALL the `pathToContent.**.name` to separate local variables with nice names

Reply:
there is a method: `com.saveourtool.save.utils.FileUtilsKt#pathNamesTill`
```suggestion
val (rootDir, ownerName, toolName, vcsTagName, executableName) = pathToContent.pathNamesTill(rootDir)
return ToolKey(rootDir, ownerName, toolName, vcsTagName, executableName)
```

Reply:
Oh wow, that might be super useful here, thanks

Reply:
BTW, @nulls, this method needs kdoc as far as it is not obvious that `rootDir` is also returned. will do it